etc/DependencyInstaller.sh

```diff
@@ -0,0 +1,132 @@
+#!/bin/bash
+
+set -euo pipefail
+
+# Make sure we are on the correct folder before beginning
+if [[ "$OSTYPE" == "darwin"* ]]; then
+    cd "$(dirname $(perl -e 'use Cwd "abs_path";print abs_path(shift)' $0))/../"
+else
+    cd "$(dirname $(readlink -f $0))/../"
+fi
+
+_installORDependencies() {
+    ./tools/OpenROAD/etc/DependencyInstaller.sh
+}
+
+_installCommon() {
+    # install pandas
+    pip3 install -U --user pandas
+}
+
+_installCentosCleanUp() {
+    yum clean -y all
+    rm -rf /var/lib/apt/lists/*
+}
+
+_installCentosPackages() {
+    yum update -y
+    yum install -y \
+        libffi-devel \
+        tcl \
+        time \
+        ruby \
+        ruby-devel \
+        tcl-devel 
+
+    yum install -y https://www.klayout.org/downloads/CentOS_7/klayout-0.27.10-0.x86_64.rpm
+}
+
+_installUbuntuCleanUp() {
+    apt-get autoclean -y
+    apt-get autoremove -y
+}
+
+_installUbuntuPackages() {
+    export DEBIAN_FRONTEND="noninteractive"
+    apt-get -y update
+    apt-get -y install \
+        libffi-dev \
+        tcl \
+        tcl-dev \
+        time \
+        ruby \
+        ruby-dev \
+        libz-dev \
+        python3-pip \
+        qttools5-dev \
+        libqt5xmlpatterns5-dev \
+        qtmultimedia5-dev \
+        libqt5multimediawidgets5 \
+        libqt5svg5-dev
+
+    lastDir="$(pwd)"
+
+    # temp dir to download and compile
+    baseDir=/tmp/installers
+    mkdir -p "${baseDir}"
+    cd ${baseDir}
+
+    # install KLayout
+    klayoutVersion=0.27.10
```

Comment:
Let's define `klayoutVersion` and any other dependency version at the top of the script file and use for all installer methods/pkg managers.